src/APIM_ARMTemplate/apimtemplate/Commands/Extract.cs

```diff
@@ -6,118 +6,129 @@
+					if (!fileFolderName.HasValue()) throw new Exception("Missing parameter <filefolder>.");
 
-                    // isolate cli parameters
-                    string resourceGroup = resourceGroupName.Values[0].ToString();
-                    string sourceApim = sourceApimName.Values[0].ToString();
-                    string destinationApim = destinationAPIManagementName.Values[0].ToString();
-                    string fileFolder = fileFolderName.Values[0].ToString();
-                    string linkedBaseUrl = linkedTemplatesBaseUrlName.HasValue() ? linkedTemplatesBaseUrlName.Value().ToString() : null;
-                    string policyXMLBaseUrl = policyXMLBaseUrlName.HasValue() ? policyXMLBaseUrlName.Value().ToString() : null;
-                    string singleApiName = null;
+					// isolate cli parameters
+					string resourceGroup = resourceGroupName.Values[0].ToString();
+					string sourceApim = sourceApimName.Values[0].ToString();
+					string destinationApim = destinationAPIManagementName.Values[0].ToString();
+					string fileFolder = fileFolderName.Values[0].ToString();
+					string linkedBaseUrl = linkedTemplatesBaseUrlName.HasValue() ? linkedTemplatesBaseUrlName.Value().ToString() : null;
+					string policyXMLBaseUrl = policyXMLBaseUrlName.HasValue() ? policyXMLBaseUrlName.Value().ToString() : null;
+					string singleApiName = null;
+					string singleRevisionName = null;
 
-                    if (apiName.Values.Count > 0)
-                    {
-                        singleApiName = apiName.Values[0].ToString();
-                    }
+					if (apiName.Values.Count > 0)
+					{
+						singleApiName = apiName.Values[0].ToString();
+					}
 
-                    Console.WriteLine("API Management Template");
-                    Console.WriteLine();
-                    Console.WriteLine("Connecting to {0} API Management Service on {1} Resource Group ...", sourceApim, resourceGroup);
-                    if (singleApiName != null)
-                    {
-                        Console.WriteLine("Executing extraction for {0} API ...", singleApiName);
-                    }
-                    else
-                    {
-                        Console.WriteLine("Executing full extraction ...", singleApiName);
-                    }
+					if (singleApiName != null && revisionName.Values.Count > 0)
+					{
+						singleRevisionName = revisionName.Values[0].ToString();
+					}
 
-                    // initialize file helper classes
-                    FileWriter fileWriter = new FileWriter();
-                    FileNameGenerator fileNameGenerator = new FileNameGenerator();
-                    FileNames fileNames = fileNameGenerator.GenerateFileNames(sourceApim);
+					Console.WriteLine("API Management Template");
+					Console.WriteLine();
+					Console.WriteLine("Connecting to {0} API Management Service on {1} Resource Group ...", sourceApim, resourceGroup);
+					if (singleRevisionName != null)
```

Comment:
The order of the conditional statements here could cause confusing WriteLine calls to be sent to the user. If I do not provide a singleApiName but I do provide a revision, it would print "Executing extraction for undefined API, Revision X", when it should be letting the user know a full extraction is being executed. I would think the order should go 1. Full extraction, 2. Single API, specific revision, 3. Single API, all revisions. Alternatively, you could change the first if statement to include singleApiName != null.

Reply:
singleRevisionName is null unless apiName is provided. I'll add singleApiName != null for readability